spaceship/lib/spaceship/two_step_client.rb

```diff
@@ -128,6 +169,12 @@ def select_device(r, device_id)
         raise ex
       end
 
+      store_session
```

Comment:
After everything finish need to store, so little refactoring that to a new function `store_session`. That let `select_device` and `handle_two_fator` to use it.